bcgov/HOUS-permit-portal
# Add support for BCeID login

[sc66675]

## Description

Adds basic BCeID omniauth login. For now, this is on the login page for testing purposes but will eventually be moved to the [reviewer invitation flow](https://app.shortcut.com/codechallengell/story/67729/add-bceid-login-to-reviewer-invitation-flow).

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

Implements Story [sc66675](https://app.shortcut.com/codechallengell/story/66675/integration-of-bceid-with-application).

## Steps to QA

1. Add the required ENV variables (see `.env.example` / [CSS dashboard](https://bcgov.github.io/sso-requests/my-dashboard/integrations) installation JSON)
2. Create a BCeID in the the appropriate environment (e.g. [DEV](https://www.development.bceid.ca/)) 
3. Navigate to the login screen, click "Login with BCeID"

Expectation: user is navigated to the BCeID login screen and upon providing correct credentials, is navigated back to the app and is in a logged in state (for now, logged in as a submitter).

-->

## UI Accessibility Checklist

The UI is temporary and will change in an upcoming PR

## General Checklist

- [x] ✅ Provide tests for your changes where relevant - N/A
- [x] 📗 Update any related documentation and include any relevant screenshots - N/A
- [x] 📱 For UI-related tasks, ensure responsive design is implemented across multiple screen size - N/A

## [optional] Are there any post-deployment tasks we need to perform?

- Update SSO Integration to support test / prod environments (pending QA / PROD environment setup)
- Add ENV variables as per SSO client details once configured
